### PR TITLE
micro: fix version reporting with micro --version

### PR DESCRIPTION
**Pull request by ibrokemypie**

### CODE REVIEW

File: srcpkgs/micro/template [hunk 1]
```diff
@@ -10,8 +10,25 @@ short_desc="A modern and intuitive terminal-based text editor"
 maintainer="Diogo Leal <diogo@diogoleal.com>"
 license="MIT"
 homepage="https://github.com/zyedidia/micro"
-distfiles="https://github.com/zyedidia/micro/archive/v${version}.tar.gz"
+distfiles="${homepage}/archive/v${version}.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/micro/template]: Not necessary.

File: srcpkgs/micro/template [hunk 2]
```diff
@@ -10,8 +10,25 @@ short_desc="A modern and intuitive terminal-based text editor"
...
+do_build() {
+	if [[ "${go_get}" != "yes" ]]; then
+		local path="${GOPATH}/src/${go_import_path}"
+		mkdir -p "$(dirname ${path})"
+		ln -fs $PWD "${path}"
+	fi
+
+	go_package=${go_package:-$go_import_path}
+	go build -ldflags "-X main.Version=${version} -X 'main.CommitHash=${_commithash}' -X 'main.CompileDate=$(date)'" -o ${GOPATH}/bin/micro ./cmd/micro
+}
```
Comment (Vaelatern) [inline srcpkgs/micro/template]: Please do not do any of these things. A simple patch or a sed line would be much much preferred.

File: srcpkgs/micro/template [hunk 3]
```diff
@@ -1,10 +1,12 @@
 # Template file for 'micro'
 pkgname=micro
 version=1.2.0
-revision=1
+revision=2
+_commithash="be8124154b2b3e761578a186c42a18ace5a43865"
 build_style=go
 go_import_path="github.com/zyedidia/micro"
 go_package="${go_import_path}/cmd/micro"
+go_ldflags="-X main.Version=${version} -X 'main.CommitHash=${_commithash}' -X 'main.CompileDate=$(date)'"
```
Comment (Vaelatern) [inline srcpkgs/micro/template]: Can we do without adding the commithash?

Comment (Vaelatern) [inline srcpkgs/micro/template]: Add a comment to the template above setting _commithash explaining the need.

Comment (ibrokemypie) [inline srcpkgs/micro/template]: have done

**Pull request closed without merging**